### DISCUSSION TITLE
New approach coming soon

### DISCUSSION THREAD

**Discussion by maheini** [maheini/FreeStyle-Libre-3-patch#6] [Announcements]
Good news!

There's a new approach and already a patched app 🥇 

I will keep track of it and add all useful informations to the repo.

Stay tuned, it looks promising.

**Comment by maheini** [comment]
I just dug deeper into it and it does in fact looks very promising. I'm currently in exchange with the developer and will soon share everything about this new solution.

**Comment by maheini** [comment]
In exchange with the developer, the Juggluco-approach has just been announced. Big and very great news for #WeAreNotWaiting.

**Comment by twinko** [comment]
@maheini 
Thank you so much for the great overview of all the solutions! Not sure where u from but if you need support with translations of german language let me know.

The nativ xdrip integration of the patched apk would be awesome, If you have more info abouzt that, please let us know!

**Reply by maheini** [reply]
Thanks for your support, but I'm from a German speaking country as well  - so there's no need right now. Or do you think, a german translation would be helpful for many people? 

It's a pleasure for me to show all those valuable informations, as I was dependent on such as well in the past. I'm just driving forward, what others have begun.

Danke für das Angebot und eine schöne Woche :)

**Comment by twinko** [comment]
@maheini btw, did you saw my pull request?

**Reply by maheini** [reply]
I did see them and will merge them when I get some time. Thanks for your contribution 😄.

**Comment by maxmillionbeatz** [comment]
🏆🏆excited for this. Thank you guys

**Comment by dassystem** [comment]
I think you are referring to the exciting "Juggluco-Solution". But I am not able to find the source code from jkaltes of his/her work. On the website I did not find a way to contact the author.

To me (and maybe others?) it seems risky to just install a binary, handling my health issues, with no (public) code review in place. I'm also mentioning this because it might be important for future adaptations and even copyright /copyleft issues.

**Reply by maheini** [reply]
AFAIK, jkaltes didn't released the source code. However, there's not a big risk involved, as the patch isn't requiring any permissions.
But in general, you're right - there's no guarantee - and never have been for Apps like patches, AAPS, Xdrip etc. It's on your own risk.

Just let me assure you that there is a contact option for jkaltes. If you got an urgent request to jkaltes, feel free to contact me. As a second guarantee: Juggluco is submitted to the Playstore, meaning it's proven by Google. Still there's no guarantee regarding the functionality, as in every other app as well. 

And finally, there shouldn't be a copyright/copyleft issue - I have been talking with jkaltes regarding this topic.

But a disclaimer is definitely a good idea for the readme. I will add it soon.

**Comment by gui-dos** [comment]


**Reply by maheini** [reply]
Thanks @gui-dos for those great insights. 👍

**Comment by drrobintravis** [comment]
Hi! I have a stable connection between the cracked libre3 and juugluco. I have an OS wear watch. I can't seem to get juugluco to send blood glucose values to the watch in any stable fashion.  It works for half an hour and then gone. I try to fiddle around best I can but no solution. Any ideas?

**Reply by twinko** [reply]
Looks like a question for a separate topic, but i guess without logs noone can help.

**Reply by maheini** [reply]
Unfortunately, I'm not able to help you a lot. I'm currently not using Juggluco (as I don't even have libre 3 sensors).

Are you using Xdrip+ (another diabetes app)? If you really need those values, I can totally recommend Xdrip -> You can connect it to Juggluco and then use the Xdrip watchfaces. It does offer a lot more features as well.

If there isn't another solution, feel free to open a Issue and submit all informations you have like phone, Android version etc.)

**Comment by dassystem** [comment]
I was able to try the patch with Juggluco myself to send the values to xdrip+. Seems to work quite well. I'm very happy, to have a offline solution. No more unnecessary data on my mobile plan, no lag between values and no more using the "official" libre link servers.

On the other side, the new menu in the libre3 app has the worst user interface in a mobile app I have seen in years. That red background color really burns into my eyes. Setting everything up was a bit of a pain, although the documentation on the website was good. I have just overlooked a single checkbox… 

But anyway: great job and I'm happy to see improvements regarding the setup/UI ;)

**Reply by 1ster** [reply]
Very happy that a solution came up on the very day I received my first pack of Libre3 sensors (despite having requested Libre2 ones).
Even cooler would be to have the patched app send directly to Xdrip+ :-)

Also, for the Libre 2 sensor, I haven't been aware that Juggluco is a perfect replacement for the (patched) Libre2 app, with the advantage that it can resume on a sensor where the Librelink app has lost connection.

**Comment by paulyglocke** [comment]
Hey everyone, so Abbott just confirmed my fear, being forced onto Libre 3, Diabox is still some way off, but now I have found this! And am thrilled! I have downloaded the patched app, and Juggluco from app store, but I would really love a clear step by step guide to get everything set up, finding the website a little confusing (http://jkaltes.byethost16.com/Juggluco/libre3/)

Also as 1ster says, direct sending of Data to Xdrip would be ideal! Thats what I was doing with Libre 2

Thanks again everyone, cant thank you enough! 

#wearenotwaiting

**Reply by twinko** [reply]
Hello @paulyglocke your nic name suggest you can speak german. Here is a pretty detailed tutorial. Its written in german. If you cant speak german, try google translater or better use deepl to translatew the text:
https://insulinclub.de/core/index.php?attachment/17545-anleitung-2-mit-links-freestyle-libre-3-und-juggluco-pdf/

**Reply by paulyglocke** [reply]
VIELEN DANK!!!!!!! I speak some german, lived here for ten years, should speak much more!!

**Reply by paulyglocke** [reply]
Also on another topic, what are the odds we will get a patched app direct to Xdrip?  Is it being worked on?

**Reply by dassystem** [reply]
> Hello @paulyglocke your nic name suggest you can speak german. Here is a pretty detailed tutorial. Its written in german. If you cant speak german, try google translater or better use deepl to translatew the text: https://insulinclub.de/core/index.php?attachment/17545-anleitung-2-mit-links-freestyle-libre-3-und-juggluco-pdf/

Since this very repository is about documenting, not about developing: Maybe we should have a translated version of this document here (and not in a seperate forum / website).

**Reply by twinko** [reply]
> Also on another topic, what are the odds we will get a patched app direct to Xdrip? Is it being worked on?

im really looking forward to such a solution. I already mentioned the repo owner of xdrip to the patched libre 3 app. He didnt responed yet and ididnt heard of anybody working on that (https://github.com/NightscoutFoundation/xDrip/discussions/2051).
But i keep my fingers crossed, there will be a solution asap :)

> Since this very repository is about documenting, not about developing: Maybe we should have a translated version of this document here (and not in a seperate forum / website).

I didnt wrote it so im not sure if i have the permission to translate and copy it over, but i think @maheini has some connections. 

As mentioned here: https://github.com/maheini/FreeStyle-Libre-3-patch/discussions/11 i could translate everything pretty quick

**Reply by maheini** [reply]
Unfortunately, I don't have any "copyright permission" on that document. I did included it into the german readme though. Thank you @twinko for your support regarding translations. I will keep it in mind and contact you if needed 😄

**Comment by master3395** [comment]
Hello, just downloaded the app.
Why does it only allow German sensors?
I am from Norway, and we just got Libre 3 approved.

![Screenshot_2022-09-20-17-59-56-87_f](https://user-images.githubusercontent.com/4199669/191307660-07309de6-7831-483e-bc82-39c6a980a48d.jpeg)

**Reply by maheini** [reply]
Hi @master3395 
It doesn't only allow german sensors - it's the german app, but you should be able to activate any kind of libre 3 sensors. Just keep Germany selected and press Next.

The german version of the app was chosen because it contains both, mmol/l and mg/dl

**Reply by master3395** [reply]
> Hi @master3395 It doesn't only allow german sensors - it's the german app, but you should be able to activate any kind of libre 3 sensors. Just keep Germany selected and press Next.
> 
> The german version of the app was chosen because it contains both, mmol/l and mg/dl

Ah thanks :)
If you need help translating the app to Norwegian I can help :)

**Reply by maheini** [reply]
Thank you as well for your offer. I will probably determine the most needed languages soon and add more translations. I will keep it in mind and contact you if needed 😄

**Comment by drrobintravis** [comment]
Thanks for all the help guys! I tried one of the suggested solutions: I
connect the watch to the mobile hotspot on the phone. The the watch works.

Den fre 16 sep. 2022 10:18Markus F. ***@***.***> skrev:
